### PR TITLE
Cache

### PR DESCRIPTION
**Pull request by bretterer**

### CODE REVIEW

File: src/Stormpath/DataStore/DefaultDataStore.php [hunk 1]
```diff
@@ -97,9 +103,18 @@ public function getResource($href, $className, array $options = array())
...
         $queryString = $this->getQueryString($options);
+
+
         $data = $this->executeRequest(Request::METHOD_GET, $href, '', $queryString);
 
-        return $this->resourceFactory->instantiate($className, array($data, $queryString));
+
+        if($this->resourceIsCacheable($data)) {
+            $this->addDataToCache($data);
+        }
```
Comment (ecrisostomo) [inline src/Stormpath/DataStore/DefaultDataStore.php]: Where are you attempting to get the resource data from the cache before executing the request to the server?

File: src/Stormpath/Cache/Cacheable.php [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+abstract class Cacheable {
+
+    protected function resourceIsCacheable($resource)
+    {
+        $cache = true;
+
+        // Check to see if it is a collection
+        // All collections will have an items array in the data
+        // If it is a collection, we do not want to cache it;
+        if(isset($resource->items)) $cache = false;
```
Comment (ecrisostomo) [inline src/Stormpath/Cache/Cacheable.php]: What about the few resources that don't have href or resources for which we don't have regions (could happen)?

File: src/Stormpath/DataStore/DefaultDataStore.php [hunk 3]
```diff
@@ -135,6 +156,10 @@ public function save(Resource $resource, $returnType = null)
 
         $returnedResource = $this->saveResource($href, $resource, $returnType);
 
+        if($this->resourceIsCacheable($returnedResource)) {
+            $this->addDataToCache($returnedResource);
```
Comment (ecrisostomo) [inline src/Stormpath/DataStore/DefaultDataStore.php]: We need to check for the account password so we don't store that information.

File: src/Stormpath/DataStore/DefaultDataStore.php (hunk 1 above)
Comment (bretterer) [inline src/Stormpath/DataStore/DefaultDataStore.php]: This was a missed commit... Pushed now.

File: src/Stormpath/Cache/Cacheable.php (hunk 2 above)
Comment (bretterer) [inline src/Stormpath/Cache/Cacheable.php]: I thought we said that all resources had a href.  Not sure what you mean by resources that do not have regions

File: src/Stormpath/DataStore/DefaultDataStore.php (hunk 3 above)
Comment (bretterer) [inline src/Stormpath/DataStore/DefaultDataStore.php]: Will add this in.

**Pull request closed without merging**